crates/ty_python_semantic/src/types.rs

```diff
@@ -1779,7 +1792,7 @@ impl<'db> Type<'db> {
                 }
             }
             Type::ClassLiteral(class_literal) => {
-                Some(class_literal.default_specialization(db).into_callable(db))
+                Some(class_literal.identity_specialization(db).into_callable(db))
```

Comment:
Is this the place where we'd want to re-bind typevars to the new callable type? Should we add a TODO for that?